e2e/consulacls/consul-acls-manage.sh

```diff
@@ -0,0 +1,375 @@
+  ssh \
+    -o StrictHostKeyChecking=no \
+    -o UserKnownHostsFile=/dev/null \
+    -i "${pemfile}" \
+    "${user}@${hostname}" "${command}"
+}
+
+function doSCP() {
+  original="$1"
+  username="$2"
+  hostname="$3"
+  destination="$4"
+  echo "------> will scp ${original} to ${hostname}"
+  scp \
+    -o StrictHostKeyChecking=no \
+    -o UserKnownHostsFile=/dev/null \
+    -i "${pemfile}" \
+    "${original}" "${username}@${hostname}:${destination}"
+}
+
+function doBootstrap() {
+  echo "=== Bootstrap: Consul Configs ==="
+
+  # Stop all Nomad agents.
+  stopNomad
+
+  # Stop all Consul agents.
+  stopConsul
+
+  # Run the activation step, which uploads the ACLs-enabled acl.hcl file
+  # to each Consul Server's configuration directory, then (re)starts each
+  # Consul Server agent.
+  doActivate
+
+  echo "=== Bootstrap: Consul ACL Bootstrap ==="
+
+  # Bootstrap Consul ACLs on server[0]
+  echo "-> bootstrap ACL using ${server0}"
+  consul_http_token=$(doSSH "${server0}" "/usr/local/bin/consul acl bootstrap" | grep SecretID | awk '{print $2}')
+  consul_http_addr="http://${server0}:8500"
+  export CONSUL_HTTP_TOKEN=${consul_http_token}
+  export CONSUL_HTTP_ADDR=${consul_http_addr}
+  echo "  consul http: ${CONSUL_HTTP_ADDR}"
+  echo "  consul root: ${CONSUL_HTTP_TOKEN}"
+  echo "${CONSUL_HTTP_TOKEN}" > "${token_file}"
+
+  # Create Consul Server Policy & Consul Server agent tokens
+  echo "-> configure consul server policy"
+  consul acl policy create -name server-policy -rules @consulacls/consul-server-policy.hcl
+
+  # Create & Set agent token for each Consul Server
+  for server in ${servers}; do
+    echo "---> will create agent token for server ${server}"
+    server_agent_token=$(consul acl token create -description "consul server agent token" -policy-name server-policy | grep SecretID | awk '{print $2}')
+    echo "---> setting token for server agent: ${server} -> ${server_agent_token}"
+    consul acl set-agent-token agent "${server_agent_token}"
+    echo "---> done setting agent token for server ${server}"
+  done
+
+  # Wait 10s before continuing with configuring consul clients.
+  echo "-> sleep 10s"
+  sleep 10
+
+  # Start the Consul Clients back up so we can set their tokens now
+  startConsulClients
+
+  # Create Consul Client Policy & Client agent tokens
+  echo "-> configure consul client policy"
+  consul acl policy create -name client-policy -rules @consulacls/consul-client-policy.hcl
+
+  # Create & Set agent token for each Consul Client (including windows)
+  for client in ${clients}; do
+    echo "---> will create consul agent token for client ${client}"
+    client_agent_token=$(consul acl token create -description "consul client agent token" -policy-name client-policy | grep SecretID | awk '{print $2}')
+    echo "---> setting consul token for consul client ${client} -> ${client_agent_token}"
+    consul acl set-agent-token agent "${client_agent_token}"
```

Comment:
I may be misunderstanding the Consul ACL docs, but doesn't this set the token for the specific agent we're talking to, and not each of the clients' agents in turn?

Reply:
Yep, really good catch! Turns out only server0 was actually being configured correctly. Added some bash shenanigans to make sure `CONSUL_HTTP_ADDR` is being set correctly for each agent now.